### ISSUE TITLE
No video on 3G

### ISSUE THREAD

**Issue by odrobnik** [Temasys/SKYLINK-iOS#6]
We found a curious problem using 1.0.11 (but earlier version no difference):

on a 4G connection or WiFi everything works without problem.
on a 3G connection, we don't receive video.

The following order of delegate calls is observed:

1. `didConnectWithMessage`
2. `didJoinPeer`
3. `didRenderPeerVideo`

but we never receive a `didChangeVideoSize` and the contents of the peer video view remains black.  It also never reflects the aspect ratio that get sent, so somehow the video stream never comes through. Also no audio is transmitted.

I also tried different max video bitrates and advanced settings to no avail, we also tried swapping the SIM card, but if the iOS app runs on the 3G network we cannot get a video. Could it be that there are certain cellular networks that block Skyline video connections? But if so, why does the SDK say that it rendered peer video?

Unfortunately this is a major show stopper for us, as we wanted to submit our iOS app to Apple within the next few days. Is there any way I could work with an engineer to find the problem and hopefully a workaround? you can contact me at oliver@cocoanetics.com

**Comment by odrobnik** [comment]
@RomainPln can you help?

**Comment by RomainPln** [comment]
Hello.
This is likely related to a recently identified scenario on mobile networks and should be fixed in next release.
I have sent you an email with a procedure that could confirm this.

**Comment by odrobnik** [comment]
I am sorry to report that the pre-release SDK did not address our issue of not getting any video transmission on the Austrian A1 cellular network. We are 100% consistent in reproducing the issue on this network.

These are the logs from the client on WiFi. You can see the `Did Render Peer` message I log in `didRenderPeerVideo`, but we never get the `didChangeVideoSize`

I notice that there are several errors and warnings in the log.

```
2017-03-27 15:38:00.652649 Klickerr[651:210172] [DYMTLInitPlatform] platform initialization successful
2017-03-27 15:40:01.666908 Klickerr[651:210074] <INFO> SkylinkSDK for iOS v_1.0.11.1
2017-03-27 15:40:01.799057 Klickerr[651:210074] [MC] System group container for systemgroup.com.apple.configurationprofiles path is /private/var/containers/Shared/SystemGroup/systemgroup.com.apple.configurationprofiles
2017-03-27 15:40:01.804952 Klickerr[651:210074] [MC] Reading from public effective user settings.
2017-03-27 15:40:01.979777 Klickerr[651:210074] <INFO> Starting local video source to the front camera
2017-03-27 15:40:02.629316 Klickerr[651:210074] <INFO> Processing the room server response ...
2017-03-27 15:40:02.630574 Klickerr[651:210074] <INFO> Connecting Sig...
2017-03-27 15:40:02.630919 Klickerr[651:210074] <INFO> Connecting to the room...
2017-03-27 15:40:02.631929 Klickerr[651:210074] <INFO> Successfully processed room info from App server
2017-03-27 15:41:14.341735 Klickerr[651:210074] <INFO> [Handshake] (restart) Closing peer connection with 'true' ...
2017-03-27 15:41:14.342653 Klickerr[651:210074] Peer NXlCQwHCYNXu06pGEm7B left: Received 'bye'
2017-03-27 15:43:21.133978 Klickerr[651:211079] <INFO> Disconnecting from the room ...
2017-03-27 15:43:23.465851 Klickerr[651:211079] <INFO> Stopped local video.
2017-03-27 15:43:30.591970 Klickerr[651:210074] <INFO> SKYLINKConnection Config => 
(SKYLINKConnectionConfig instance description):
audio:'true', video:'true', dataChannel:'false', fileTransfer:'false', timeout:'60', disableTURN:'false', disableSTUN:'false', disableHOST:'false' prefered transport:'(null)', prefered audioCodec:'(null)', useBackCamera (for local video, if video is enabled):'false', prefered session presets:'(
    AVCaptureSessionPreset640x480,
    AVCaptureSessionPreset352x288
)', receiveVideo:'true', receiveVideo:'true', maxAudioBitrate:'0', maxVideoBitrate:'512', maxDataBitrate:'0'.
2017-03-27 15:43:30.600373 Klickerr[651:210074] <INFO> SkylinkSDK for iOS v_1.0.11.1
2017-03-27 15:43:30.697433 Klickerr[651:210074] <INFO> Starting local video source to the front camera
2017-03-27 15:43:31.623677 Klickerr[651:210074] <INFO> Processing the room server response ...
2017-03-27 15:43:31.624736 Klickerr[651:210074] <INFO> Connecting Sig...
2017-03-27 15:43:31.624871 Klickerr[651:210074] <INFO> Connecting to the room...
2017-03-27 15:43:31.625713 Klickerr[651:210074] <INFO> Successfully processed room info from App server
2017-03-27 15:43:34.083245 Klickerr[651:210991] dnssd_clientstub write_all(15) DEFUNCT
2017-03-27 15:43:34.083315 Klickerr[651:210991] dnssd_clientstub deliver_request ERROR: write_all(15, 60 bytes) failed
2017-03-27 15:43:34.083400 Klickerr[651:210991] dnssd_clientstub write_all(15) DEFUNCT
2017-03-27 15:51:22.692745 Klickerr[651:210074]  ⚠️ <WARNING>  ► [Handshake] notice in welcome since PC exists.
2017-03-27 15:51:22.693037 Klickerr[651:210074] <INFO> -Welcome-handle
2017-03-27 15:51:22.693313 Klickerr[651:210074] <INFO> Processing [W]
2017-03-27 15:51:25.857955 Klickerr[651:210074] Did Render Peer HpykUyA9hnfX4ljXEnbo
2017-03-27 15:51:25.858147 Klickerr[651:210074] Showing Video for Peer HpykUyA9hnfX4ljXEnbo
2017-03-27 15:51:25.890778 Klickerr[651:210074] Failed to bind EAGLDrawable: <CAEAGLLayer: 0x5755400> to GL_RENDERBUFFER 1
2017-03-27 15:51:25.890898 Klickerr[651:210074] Failed to make complete framebuffer object 8cd6
2017-03-27 15:51:33.770086 Klickerr[651:212123] <INFO> Disconnecting from the room ...
2017-03-27 15:51:33.774024 Klickerr[651:212123] <INFO> [Handshake] (restart) Closing peer connection with 'false' ...
2017-03-27 15:51:35.297449 Klickerr[651:212123] <INFO> Stopped local video.
2017-03-27 15:51:37.215183 Klickerr[651:210074] <INFO> SKYLINKConnection Config => 
(SKYLINKConnectionConfig instance description):
audio:'true', video:'true', dataChannel:'false', fileTransfer:'false', timeout:'60', disableTURN:'false', disableSTUN:'false', disableHOST:'false' prefered transport:'(null)', prefered audioCodec:'(null)', useBackCamera (for local video, if video is enabled):'false', prefered session presets:'(
    AVCaptureSessionPreset640x480,
    AVCaptureSessionPreset352x288
)', receiveVideo:'true', receiveVideo:'true', maxAudioBitrate:'0', maxVideoBitrate:'512', maxDataBitrate:'0'.
2017-03-27 15:51:37.222488 Klickerr[651:210074] <INFO> SkylinkSDK for iOS v_1.0.11.1
2017-03-27 15:51:37.309504 Klickerr[651:210074] Failed to bind EAGLDrawable: <CAEAGLLayer: 0x146bf260> to GL_RENDERBUFFER 1
2017-03-27 15:51:37.309612 Klickerr[651:210074] Failed to make complete framebuffer object 8cd6
2017-03-27 15:51:37.311106 Klickerr[651:210074] <INFO> Starting local video source to the front camera
2017-03-27 15:51:38.251963 Klickerr[651:210074] <INFO> Processing the room server response ...
2017-03-27 15:51:38.252797 Klickerr[651:210074] <INFO> Connecting Sig...
2017-03-27 15:51:38.252932 Klickerr[651:210074] <INFO> Connecting to the room...
2017-03-27 15:51:38.253833 Klickerr[651:210074] <INFO> Successfully processed room info from App server
2017-03-27 15:51:40.924209 Klickerr[651:210074] <INFO> [Handshake] (restart) Closing peer connection with 'true' ...
2017-03-27 15:51:40.924717 Klickerr[651:210074] Peer HpykUyA9hnfX4ljXEnbo left: Received 'bye'
```

**Comment by RomainPln** [comment]
fixed in 1.0.11.1 confirmed by email.
